sources/packages/web/src/components/common/IdleTimeChecker.vue

```diff
@@ -57,44 +84,106 @@ export default {
       clearInterval(interval.value);
       if (isAuthenticated.value) {
-        /* eslint-disable */
-        interval.value = setInterval(checkIdle, 30000);
-        /* eslint-enable */
+        interval.value = setInterval(checkIdle, IDLE_CHECKER_TIMER_INTERVAL);
       }
     };
 
+    watch(
+      () => extendTimeModal.value.showDialog,
+      (newValue: boolean) => {
+        if (newValue) {
+          countdownInterval = setInterval(() => {
+            if (countdown.value > 0) {
+              countdown.value = maximumIdleTime.value - idleTimeInSeconds;
+            }
+          }, COUNTDOWN_INTERVAL);
+        } else {
+          clearInterval(countdownInterval);
+          countdown.value = COUNT_DOWN_TIMER_FOR_LOGOUT;
+        }
+      },
+    );
+
+    const extendUserSessionTime = () => {
+      setLastActivityTime();
+      resetIdleCheckerTimer();
+      extendTimeModal.value.showDialog = false;
+      clearInterval(countdownInterval);
+      countdown.value = COUNT_DOWN_TIMER_FOR_LOGOUT;
+    };
+
     const confirmExtendTimeModal = async () => {
+      extendTimeModal.value.showDialog = true;
       if (await extendTimeModal.value.showModal()) {
-        lastActivityLogin.value = new Date();
-        resetIdleCheckerTimer();
+        extendUserSessionTime();
+      } else {
+        setLoggedOut();
       }
     };
 
     const checkIdle = () => {
-      const idleTimeInMintutes = getDatesDiff(
-        lastActivityLogin.value,
+      if (getLoggedOut()) {
+        logoff();
+        return;
+      }
+      idleTimeInSeconds = getDatesDiff(
+        getLastActivityTime(),
         new Date(),
         "second",
-        true,
+        false,
       );
-      if (idleTimeInMintutes >= minimumIdleTime.value) {
+
+      // Exceeded user session time.
+      if (idleTimeInSeconds > maximumIdleTime.value) {
+        // Logoff immediately in case session is expired.
+        logoff();
+      } else if (
+        idleTimeInSeconds >=
+        maximumIdleTime.value - COUNT_DOWN_TIMER_FOR_LOGOUT
+      ) {
         confirmExtendTimeModal();
       }
     };
 
-    onMounted(async () => {
-      resetIdleCheckerTimer();
-    });
-
     const setLastActivityTime = () => {
       if (isAuthenticated.value) {
-        lastActivityLogin.value = new Date();
+        localStorage.setItem(
+          LAST_ACTIVITY_TIME_LOCAL_STORAGE_ITEM,
+          new Date().getTime().toString(),
+        );
+      }
+    };
+
+    const getLastActivityTime = () => {
+      const epochLastActivityTimeStringValue = localStorage.getItem(
+        LAST_ACTIVITY_TIME_LOCAL_STORAGE_ITEM,
+      );
+      if (epochLastActivityTimeStringValue) {
+        return new Date(+epochLastActivityTimeStringValue);
       }
+      return new Date();
     };
+
+    const setLoggedOut = () => {
+      localStorage.setItem(LOGGED_OUT_LOCAL_STORAGE_ITEM, "true");
+    };
+
+    const getLoggedOut = () => {
+      const loggedOutStringValue = localStorage.getItem(
```

Comment:
In the same way that the method `setLoggedOut` was added to the `useAuth` can we add this also? In this way the only way to manipulate the `LOGGED_OUT_LOCAL_STORAGE_ITEM` would be from the  `useAuth` and event the const LOGGED_OUT_LOCAL_STORAGE_ITEM can be set only there.

Reply:
Yes, you're right. I'll change that. Thanks!

Reply:
Done.